limit = 15

def fib(first_num, second_num)
  while first_num + second_num < limit
    sum = first_num + second_num
    first_num = second_num
    second_num = sum
  end
  sum
end

result = fib(0, 1)
puts "result is #{result}"

=begin

The error is caused by variable scope. Methods contain their own variable scope
and therefore cannot access local variable `limit` as it is outside the scope of
the method. In order for this local variable to be accessible, it would either
need to be represented by a constant or must be included in the method's parameter
and passed in as an argument.

Example 1:

LIMIT = 15

def fib(first_num, second_num)
  while first_num + second_num < LIMIT
    sum = first_num + second_num
    first_num = second_num
    second_num = sum
  end
  sum
end

result = fib(0, 1)
puts "result is #{result}"

Example 2:

limit = 15

def fib(first_num, second_num, limit)
  while first_num + second_num < limit
    sum = first_num + second_num
    first_num = second_num
    second_num = sum
  end
  sum
end

result = fib(0, 1, limit))
puts "result is #{result}"

=end